src/components/signup-form/SignupForm.tsx

```diff
@@ -0,0 +1,143 @@
+import React from 'react';
+import { Button, Col, Form, Input, Row, Typography } from 'antd';
+import { ParagraphProps } from 'antd/lib/typography/Paragraph';
+import { Link } from 'react-router-dom';
+import { TEXT_GREY } from '../../colors';
+import styled from 'styled-components';
+import useWindowDimensions, {
+  WindowTypes,
+} from '../../components/window-dimensions';
+
+const { Paragraph } = Typography;
+
+const formHalfItemSpan = 11;
+const offsetSpan = 1;
+
+const formLayout = {
+  wrapperCol: { span: 17 },
+};
+
+const Footer: typeof Paragraph = styled(Paragraph)<ParagraphProps>`
+  color: ${TEXT_GREY};
+  line-height: 1.5;
+`;
+
+interface SignupFormProps {
+  readonly onFinish: any;
+}
+
+interface ButtonFooterProps {
+  readonly isMobile: boolean;
+}
+
+const ButtonFooter: React.FC<ButtonFooterProps> = ({ isMobile }) => {
+  const SignupButton = (
+    <Button type="primary" htmlType="submit" size="large">
+      Sign Up
+    </Button>
+  );
+  const FooterToLogin = (
+    <Footer>
+      ALREADY HAVE AN ACCOUNT?
+      <br />
+      LOGIN <Link to="/login">HERE!</Link>
+    </Footer>
+  );
```

Comment:
Typically we should avoid having JSX outside of the return statement of a component, I think it makes more sense here to include these in the return, duplicated code is less of an issue.